account_banking_pain_base/models/banking_export_pain.py

```diff
@@ -299,14 +299,16 @@ def generate_party_agent(
                 party_agent_institution, gen_args.get('bic_xml_tag'))
             party_agent_bic.text = bic
         except Warning:
-            if order == 'C':
-                if iban[0:2] != gen_args['initiating_party_country_code']:
-                    raise Warning(
-                        _('Error:'),
-                        _("The bank account with IBAN '%s' of partner '%s' "
-                            "must have an associated BIC because it is a "
-                            "cross-border SEPA operation.")
-                        % (iban, party_name))
+            # ESBEP-26920: The BIC code is not required (anymore) for cross
```

Comment:
Did you try checking the BIC instead of the try? I'm thinking going through these lines twice adds a little overhead the the already overweight process.

Reply:
Thanks for explaining that `bic` is an expression that is evaluated in the next call!